test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java

```diff
@@ -1815,4 +1815,45 @@ public void testDeleteObjectWithSubobject() throws Exception {
 
         assertThat("objects not deleted", res2, is(set()));
     }
+    @Test
```

Comment:
If this is the test we discussed previously, the case I was worried about was this:

1) Index an object with subobjects (so records in 2 indexes would be updated)
2) Index a new version of the object that either has a different or missing SearchType for the main object or the subobjects or both (imagine saving over a genome with reads, for example)
3) Would that cause the records of the older version of the objects to have `islast` set to false?

Reply:
Because the storage object type is different. A genome will have a different (set of) search type than reads. Reads probably won't have a sub object type at all, while a genome will have GenomeFeatures.

Reply:
For example:
```python
In [11]: ws.create_workspace({'workspace': 'dztest'})
Out[11]: 
[35854,
 u'dztest',
 u'gaprice',
 u'2018-08-22T20:41:31+0000',
 0,
 u'a',
 u'n',
 u'unlocked',
 {}]

In [12]: ws.save_objects({'workspace': 'dztest', 'objects': [{'type': 'Empty.AType-1.0', 'data': {'foa': 'bar'}, 'name': 'dz'}]})
Out[12]: 
[[1,
  u'dz',
  u'Empty.AType-1.0',
  u'2018-08-22T20:41:37+0000',
  1,
  u'gaprice',
  35854,
  u'dztest',
  u'c68986e24d15817eac2b6e98b1dad6a7',
  13,
  {}]]

In [13]: ws.save_objects({'workspace': 'dztest', 'objects': [{'type': 'Empty.ARef-1.0', 'data': {'ref': 'dztest/dz'}, 'name': 'dz'}]})
Out[13]: 
[[1,
  u'dz',
  u'Empty.ARef-1.0',
  u'2018-08-22T20:41:42+0000',
  2,
  u'gaprice',
  35854,
  u'dztest',
  u'ba99531e85ba40b05ee9a09ffa31dd62',
  19,
  {}]]
```

These are obviously just test types, but they demonstrate that a new version of an object could have a completely different workspace type, and therefore a completely different set of search types.

Reply:
wait why they considered different versions of an object if their types are different? Is the versions solely based on the name of the object??

Reply:
so if I had an e coli genome and I named it to rhodobacter.reads and then added the actual rhodobacter reads, the e coli genome would be overridden by the reads?

Reply:
In that case, your suspicions are correct. If the two objects on different search types, the first ver object will not have isLast set to false.